### PR TITLE
Feature/476 llm test generation

### LINKED ISSUE #467: [FEATURE] LLM Feature Store Integration — Embed features for ML pipelines
## Description

Integrate LLM-generated features (embeddings, scores, explanations) into the
AstroML feature store for use in ML pipelines.

## Scope

Connect LLM outputs to the feature store for downstream consumption.

## Files to Touch/Create

- `astroml/features/llm_features.py` — LLM feature definitions
- `astroml/features/embedding_features.py` — Embedding features
- `astroml/features/scoring_features.py` — LLM score features
- `astroml/features/llm_generators.py` — Feature generation
- `astroml/llm/features/integration.py` — Integration layer
- `astroml/llm/features/compute.py` — Feature computation
- `astroml/features/pipeline.py` — Update to include LLM features

## Feature Types

1. **Embedding Features**:
   - Transaction description embeddings
   - Account behavior embeddings
   - Alert description embeddings

2. **Score Features**:
   - Fraud probability from LLM
   - Explanation confidence score
   - Uncertainty estimates

3. **Meta Features**:
   - Prompt version used
   - Model used
   - Latency attribution

## Implementation Details

- Feature computation as part of pipeline
- Materialized views for fast access
- TTL and refresh policies
- Feature versioning
- Backfill support for historical data

## Acceptance Criteria

- LLM features available in feature store
- Features updated in real-time pipeline
- Backfill completes in <24h
- Feature quality matches standalone LLM calls
- Downstream models can use LLM features
- Feature versioning supports rollback

## Integration Points

- Fraud detection model can use LLM scores
- Recommendation system uses embeddings
- Alert prioritization uses LLM confidence

## Labels

`enhancement`, `llm`, `feature-store`, `integration`

### LINKED ISSUE #468: [FEATURE] LLM Fine-tuning Pipeline — Custom model fine-tuning for domain-specific tasks
## Description

Build infrastructure for fine-tuning LLMs on AstroML-specific data to improve
performance on domain tasks like fraud explanation, transaction classification, and SQL generation.

## Scope

Create fine-tuning pipeline with data preparation, training orchestration, and model deployment.

## Files to Touch/Create

- `astroml/llm/fine_tuning/__init__.py`
- `astroml/llm/fine_tuning/pipeline.py` — Fine-tuning orchestration
- `astroml/llm/fine_tuning/dataset.py` — Dataset preparation and formatting
- `astroml/llm/fine_tuning/trainer.py` — Training wrapper (OpenAI, Anthropic, LoRA)
- `astroml/llm/fine_tuning/evaluator.py` — Post-tuning evaluation
- `astroml/llm/fine_tuning/registry.py` — Fine-tuned model registry
- `astroml/data/fine_tuning/` — Training data storage
- `configs/llm/fine_tuning/` — Fine-tuning configs

## Fine-tuning Targets

1. **Fraud Explanation Model**: Better at explaining Stellar fraud patterns
2. **SQL Generation Model**: Improved NL-to-SQL for AstroML schema
3. **Transaction Classification**: Categorize transaction types accurately
4. **Support Chatbot**: Domain-specific customer support

## Implementation Details

- Support OpenAI fine-tuning API
- Support LoRA/QLoRA for open-source models (Llama, Mistral)
- Dataset versioning and lineage tracking
- Automated evaluation against holdout sets
- A/B testing: fine-tuned vs base model
- Rollback capability

## Acceptance Criteria

- Fine-tuned models show >15% improvement on domain tasks
- Training pipeline runs unattended
- Model evaluation compares to baseline automatically
- Fine-tuned models deployable via model registry
- Dataset preparation is reusable across tasks
- Cost tracking per fine-tuning run

## Dataset Requirements

- Minimum 500 examples per task
- 80/10/10 train/validation/test split
- Data quality validation (no duplicates, correct format)
- Synthetic data generation support

## Labels

`enhancement`, `llm`, `fine-tuning`, `ml`

### LINKED ISSUE #476: [FEATURE] LLM-powered Test Generation — Automatic test case creation from requirements
## Description

Use LLMs to automatically generate test cases, test data, and test assertions
from code, documentation, and requirements.

## Scope

Build AI-powered testing assistant that creates comprehensive test suites.

## Files to Touch/Create

- `astroml/llm/testing/__init__.py`
- `astroml/llm/testing/generator.py` — Test generation orchestrator
- `astroml/llm/testing/strategies/` — Generation strategies
- `astroml/llm/testing/assertions/` — Assertion generators
- `astroml/llm/testing/fixtures/` — Test data generation
- `astroml/llm/testing/reviewer.py` — Test quality reviewer
- `tools/test_generator/` — CLI tool for test generation

## Test Generation Capabilities

1. **Unit Tests**: Generate from function signatures and docstrings
2. **Integration Tests**: Generate from API specs
3. **Property-based Tests**: Generate from type specifications
4. **Edge Case Discovery**: Identify boundary conditions
5. **Regression Tests**: Generate from bug reports
6. **Performance Tests**: Generate load test scenarios

## Implementation Details

- Analyze code structure and documentation
- Generate pytest-compatible test functions
- Create realistic test fixtures and mocks
- Include edge cases and error conditions
- Generate test documentation
- Suggest test coverage improvements

## Acceptance Criteria

- Generated tests pass 80% of the time initially
- Human review reduces fixing time by 50%
- Test coverage suggestions are actionable
- Generated edge cases find real bugs
- Supports pytest, unittest, and FastAPI TestClient
- Test quality improves with feedback

## Example Output

```python
def test_fraud_score_with_high_risk_account():
    account = create_test_account(risk_score=0.9)
    result = await fraud_service.score(account.id)
    assert result.score > 0.8
    assert result.alert_created is True
```

## Labels

`enhancement`, `llm`, `testing`, `developer-tools`

### LINKED ISSUE #574: #79: Implement Plugin Architecture for Feature Computers
**Severity**: Medium  
**Component**: `astroml/features/`  
**Description**: Feature computers are hard-coded; adding new ones requires editing `feature_store.py`.

### Files to modify

- `astroml/features/`
- Related test files in tests/
- Configuration files if applicable
- Documentation files as needed

### Out of scope

- Refactoring unrelated modules
- Changing public API contracts
- Major version bumps or breaking changes
- Performance optimizations not directly related to the issue

### Acceptance criteria

- [ ] Code changes implemented as per procedure steps
- [ ] All new code has complete type hints
- [ ] Unit tests added/updated with >90% coverage for changed code
- [ ] All tests pass: pytest tests/
- [ ] Linting passes: ruff check . && black --check .
- [ ] Type checking passes: mypy astroml/
- [ ] Documentation updated if needed
- [ ] Changes reviewed and approved

### Procedure

1. Create entry point group in `pyproject.toml`:
   ```toml
   [project.entry-points."astroml.feature_computers"]
   frequency = "astroml.features.frequency"
   structural = "astroml.features.structural_importance"
   ```
2. Implement plugin discovery in `FeatureRegistry`:
   ```python
   from importlib.metadata import entry_points
   for ep in entry_points(group="astroml.feature_computers"):
       self.register_computer(ep.name, ep.load())
   ```
3. Add plugin registration API
4. Document how to create external feature computer plugins
5. Add example external plugin in `examples/plugin_example.py`
6. Add plugin validation (check required methods)
7. Run load tests with multiple plugins registered

### PR DESCRIPTION
**Pull request by oraimoitel**
Closes #468

---

Closes #574

---

Closes #467

---

Closes #476

---

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by oraimoitel** [comment]
Closes #580
